site/content/en/docs/Tasks/labels_and_annotations.md

```diff
@@ -2,7 +2,293 @@
 title: "Labels and Annotations"
 linkTitle: "Labels and Annotations"
 weight: 3
-date: 2017-01-05
+date: 2023-10-14
 description: >
   Working with Labels and Annotations
 ---
+
+A common set of labels can be applied to all Resources in a project by adding a [`labels`] or [`commonLabels`] entry to the `kustomization.yaml` file. Similarly, a common set of annotations can be applied to Resources with the [`commonAnnotations`] field.
+
+## Working with Labels
+### Add Labels
+[`labels`] can be used to add labels to the `metadata` field of all Resources in a project. This will override values for label keys that already exist.
+
+Here is an example of how to add labels to the `metadata` field.
+1. Create a Kustomization file.
+```yaml
+# kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+labels:
+- pairs:
+    someName: someValue
+    owner: alice
+    app: bingo
+
+resources:
+- deploy.yaml
+- service.yaml
+```
+
+2. Create Deployment and Service manifests.
+```yaml
+# deploy.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example
+```
+```yaml
+# service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: example
+```
+
+3. Add labels with `kustomize build`.
+```bash
+kustomize build .
+```
+
+The output shows that the `labels` field is used to add labels to the `metadata` field of the Service and Deployment Resources.
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: bingo
+    owner: alice
+    someName: someValue
+  name: example
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: bingo
+    owner: alice
+    someName: someValue
+  name: example
+```
+### Add Template Labels
+[`labels.includeTemplates`] can be used to add labels to the `spec.template` field of all applicable Resources in a project.
```

Comment:
Can we change "`spec.template`" to "template"?

As you can see from this file that lists all the template fields https://github.com/kubernetes-sigs/kustomize/blob/master/api/internal/konfig/builtinpluginconsts/metadatalabels.go, not all template fields have field path `spec.template`. In fact, not all template fields have field name `template`. Sorry I didn't catch this earlier.

Reply:
Good catch. Thank you for the clarification and for sharing the code reference.